### PR TITLE
chore(ci): Automated release creation

### LINKED ISSUE #390: Automate regular chores (SLS plugin)
This repo has no automation around updating semantic version, and no automation to update test fixtures, resulting in repetitive work with each release.

### PR DESCRIPTION
**Pull request by mrickard**
Changed title of main workflow, to be standardized with other repos.
Added NEWS.md release history
Added release prep, creation, and pr validation workflows from aws-sdk repo

Closes #390 
Closes NR-184573

### CODE REVIEW

File: .github/workflows/create-release.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+name: Create Release
+
+on: [workflow_dispatch]
+
+jobs:
+  tag-and-publish:
+    uses: newrelic/node-newrelic/.github/workflows/release-creation.yml@main
+    with:
+      changelog_file: NEWS.md
```
Comment (bizob2828) [inline .github/workflows/create-release.yml:9]: can we change this to CHANGELOG.md and update references to NEWS.md?

File: .github/workflows/prepare-release.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+          - patch
+          - minor
+          - major
+
+jobs:
+  release-notes:
+    uses: newrelic/node-newrelic/.github/workflows/prep-release.yml@main
+    with:
+      release_type: ${{ github.event.inputs.release_type }}
+      use_new_release: ${{ vars.USE_NEW_RELEASE }}
```
Comment (bizob2828) [inline .github/workflows/prepare-release.yml:21]: you need to set this in action variables settings

File: NEWS.md [hunk 3]
```diff
@@ -0,0 +1,422 @@
+#### Features
```
Comment (bizob2828) [inline NEWS.md:1]: I think these headers aren't needed

**Review by bizob2828 — CHANGES_REQUESTED** [review]
Overall looks fine and hard to test until we actually have to release. One comment on changelog but you'll need a few more things:
 * I would add a pull request template
 * add dependabot.yml to control conventional commits for its Prs
 
You can see examples [here](https://github.com/newrelic/node-newrelic-aws-sdk/pull/232/files#diff-18813c86948efc57e661623d7ba48ff94325c9b5421ec9177f724922dd553a35)

Also you're going to want to update the pull request settings:
 * only squash merges and use pull request title
 * delete head branches


screenshot 2024-03-20 at 9 19 13 AM

File: .github/workflows/create-release.yml (hunk 1 above)
Comment (mrickard) [reply .github/workflows/create-release.yml:9]: Sure--no problem there.

**Comment by mrickard** [comment]
@bizob2828 I've updated the PR settings as well as adding templates.

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**